src/anim/evaluator/anim-target-value.js

```diff
@@ -0,0 +1,69 @@
+import { Quat } from '../../math/quat.js';
+import { Vec3 } from '../../math/vec3.js';
+import { ANIM_LAYER_OVERWRITE } from '../controller/constants.js';
+
+class AnimTargetValue {
+    constructor(component, type) {
+        this._component = component;
+        this.mask = new Int8Array(component.layers.length);
+        this.weights = new Float32Array(component.layers.length);
+        this.totalWeight = 0;
+        this.counter = 0;
+        this.layerCounter = 0;
+        this.valueType = type;
+
+        if (this.valueType === AnimTargetValue.TYPE_QUAT) {
+            this.value = new Quat();
+            this._currentValue = new Quat();
+        } else {
+            this.value = new Vec3();
+            this._currentValue = new Vec3();
+        }
+    }
+
+    weight(index) {
+        if (this._component.dirtyWeights) this.updateWeights();
```

Comment:
is dirtyWeights getting set to false anywhere? I don't see it in the component .. but maybe elsewhere it does. Just making sure.